opts/opts.go

```diff
@@ -254,12 +254,23 @@ func validateDomain(val string) (string, error) {
 	return "", fmt.Errorf("%s is not a valid domain", val)
 }
 
-// ValidateLabel validates that the specified string is a valid label, and returns it.
+// ValidateLabel validates that the specified string is a valid label,
+// it does not use the reserved namespaces com.docker.*, io.docker.*, org.dockerproject.*
+// and returns it.
 // Labels are in the form on key=value.
 func ValidateLabel(val string) (string, error) {
 	if strings.Count(val, "=") < 1 {
 		return "", fmt.Errorf("bad attribute format: %s", val)
 	}
+
+	lowered := strings.ToLower(val)
+	if strings.HasPrefix(lowered, "com.docker.") || strings.HasPrefix(lowered, "io.docker.") ||
```

Comment:
Is this `ValidateLabel` only used by the daemon config? Double checking because if this one is also used in the CLI, we can't reject the label here (because the `docker` cli will set labels with this prefix (e.g. when doing `docker stack deploy`, the docker cli will set `com.docker.xx` labels based on the stack that services, networks and so on are part of 🤔 )

Reply:
Thanks. I have checked.  CLI has its own `ValidateLabel` function. https://github.com/docker/cli/blob/d443b7409192f29bf82258396f0170dff6cba707/opts/opts.go#L280

Reply:
yes, I agree.